Lib9c/Model/Item/Inventory.cs

```diff
@@ -442,7 +442,33 @@ public static bool IsMaterialRemovable(Item item, int id, long blockIndex)
         }
 
         /// <summary>
-        /// Remove a material from the inventory.
+        /// Checks if the given item is non-tradable material and can be removed from the inventory.
+        /// </summary>
+        /// <param name="item">The item to check for removal.</param>
+        /// <param name="id">The ID of the material to remove.</param>
+        /// <returns>Returns a boolean indicating whether the item is removable.</returns>
+        public static bool IsMaterialRemovable(Item item, int id)
+        {
+            if (item.Locked)
+            {
+                return false;
+            }
+
+            if (item.item is Material material && material.Id == id)
```

Comment:
`material.Id == id` seems not related to *'is removable'*. I think it should be like the below example because `id` is defined in `ItemBase`, not only for `Material`.

```csharp
List<Item> targetItems = _items.Where(item => item.Id == id && IsRemovableNonTradableMaterial(item)).ToList();
```

Or more..

```csharp
bool IsNonTradableMaterial(ItemBase item) => item is Material && item is not TradableMaterial;
List<Item> targetItems = _items.Where(item => !item.Locked && item.item.Id == id && IsNonTradableMaterial(item.item)).ToList();
```